adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -689,15 +691,19 @@ def socket_connected(self, socket_num):
         """Test if a socket is connected to the destination, returns boolean true/false"""
         return self.socket_status(socket_num) == SOCKET_ESTABLISHED
 
-    def socket_write(self, socket_num, buffer):
+    def socket_write(self, socket_num, buffer, conn_mode=TCP_MODE):
         """Write the bytearray buffer to a socket"""
         if self._debug:
             print("Writing:", buffer)
         self._socknum_ll[0][0] = socket_num
         sent = 0
-        for chunk in range((len(buffer) // 64) + 1):
```

Comment:
Why did you move the loop to L704? Could you add the conditionals for UDP/TCP connection mode above the loop?

Reply:
I didn't move the loop, I added code above it, to extract the loop range bound to a variable, and to select which command to send per-loop.

I think the diff isn't as clear because L698 in the old matches L704 in the new, but the variable extraction makes it look like a delete/add in the noise.